src/teleport.cpp

```diff
@@ -0,0 +1,92 @@
+#include "teleport.h"
+
+#include "character.h"
+#include "avatar.h"
+#include "creature.h"
+#include "player.h"
+#include "game.h"
+#include "map.h"
+#include "messages.h"
+#include "point.h"
+
+
+const efftype_id effect_teleglow( "teleglow" );
+
+bool teleport::teleport( Creature *const critter, int min_distance, int max_distance, bool safe,
+                         bool add_teleglow )
+{
+    if( critter == nullptr || min_distance > max_distance ) {
+        debugmsg( "ERROR: Function teleport::teleport called with invalid arguments." );
+        return false;
+    }
+
+    bool c_is_u = ( critter == &g->u );
+    player *const p = critter->as_player();
+    int tries = 0;
+    tripoint origin = critter->pos();
+    tripoint new_pos = tripoint_zero;
+    do {
+        int rangle = rng( 0, 360 );
+        int rdistance = rng( min_distance, max_distance );
+        new_pos.x = origin.x + rdistance * cos( rangle );
+        new_pos.y = origin.y + rdistance * sin( rangle );
+        tries++;
+    } while( g->m.impassable( new_pos ) && tries < 20 );
+    //handles teleporting into solids.
+    if( g->m.impassable( new_pos ) ) {
+        if( safe ) {
+            if( c_is_u ) {
+                add_msg( m_bad, _( "You cannot teleport safely" ) );
+            }
+            return false;
+        } else {
```

Comment:
The else branch is not needed as you return from the other branch above.
```
if( ... ) {
    ...
    return;
}
... // the code that used to be in the else-branch
```
That way you can remove one level of indentation (which means less white space on the screen).